### DISCUSSION TITLE
zkp中，verifer的逻辑是由什么决定的？

### DISCUSSION THREAD

**Discussion by weixiecui** [coset-io/zkp-academy#15] [Q&A]
比如halo2（kzg）为例子，链上有verifier contract。 
问verifier的逻辑是由什么决定的，想想成一个函数，输入xxx和yyy得到这个verifier contract。  
那这个contract由什么确定。我能想到的是1.使用的多项式承诺。 2.ecc的加密算法（椭圆曲线，p和生成元)。  
这里的这些正确吗，覆盖全面吗？

**Comment by luckyyang** [comment]
verifer的逻辑可以看做是验证 pairing 的函数, 不同的电路生成的多项式和 commitment 可以看做是传入这个 pairing 函数的参数
